lab-taylor/lib/player-crud.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+const debug = require('debug')('soccer:player-crud');
+const Player = require('../model/player');
+const teamCrud = require('../lib/team-crud');
+const AppError = require('./AppError');
+
+exports.createPlayer = function (reqBody) {
+  debug('creating player');
+  return new Promise((resolve, reject) => {
+    if (!reqBody.name) return reject(AppError.error400('players require a name'));
+    if (!reqBody.hometown) return reject(AppError.error400('players require a hometown'));
+    if (!reqBody.position) return reject(AppError.error400('players require a position'));
+    if (!reqBody.number) return reject(AppError.error400('players require a number'));
+    if (!reqBody.teamId) return reject(AppError.error400('players need to have a team'));
+
+    reqBody.timestamp = new Date();
+    const player = new Player(reqBody);
+    player.save()
+    .then(resolve)
+    .catch(reject);
+  });
+};
+
+exports.fetchPlayer = function(id) {
+  debug('fetching player');
+  return new Promise((resolve, reject) => {
+    Player.findOne({_id: id})
+    .then(resolve)
+    .catch(err => reject(AppError.error404(err.message)));
+  });
+};
+
+exports.updatePlayer = function(id, reqBody) {
+  debug('updating player');
+  return new Promise((resolve, reject) => {
+    if (Object.keys(reqBody).length === 0) return reject(AppError.error400('need to provide a body'));
+
+    const playerKeys = ['name', 'hometown', 'number', 'position', 'teamId'];
+    Object.keys(reqBody).forEach((key) => {
+      if (playerKeys.indexOf(key) === -1) return reject(AppError.error400('need to update a key that exists'));
+    });
+
+    Player.findByIdAndUpdate(id, reqBody)
```

Comment:
Not sure if I commented on this with you before, but there is an option you can pass in for .findByIdAndUpdate() after the "content" argument, {new: true}. This will return the updated model so that you don't have to look it up again to get the updated version.